joeynmt/training.py

```diff
@@ -75,6 +75,10 @@ def __init__(self, model: Model, config: dict) -> None:
         self.optimizer = build_optimizer(config=train_config,
                                          parameters=model.parameters())
 
+        # save checkpoint by epoch of training - used in case of no valid_data
+        self.checkpoint_epochfreq = train_config.get("checkpoint_epochfreq", 10000)
```

Comment:
Is it saved every 10000 epochs? or iterations? Maybe make naming more clear, e.g. "save_freq" if it's like "validation_freq"

Reply:
What if "validation_freq" and "checkpoint_epochfreq" are set?

Reply:
name changed to save_freq as suggested with default value of -1

If there is dev_set with positive validation_freq and save_freq, called checkpoint_epochfreq previously, checkpoints are saved when they meet the corresponding requirements. The checkpoints are managed under the same queue. Hence, a larger queue size may be required

Reply:
Cool!